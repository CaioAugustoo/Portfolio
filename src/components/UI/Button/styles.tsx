import styled, { css, DefaultTheme } from "styled-components";
import { ButtonProps } from "types/types";

const buttonModifiers = {
  isOutlined: (theme: DefaultTheme) => css`
    background-color: transparent;
    border: 1px solid ${theme.colors.primary};
    color: ${theme.colors.white};
  `,
};

export const Button = styled.button<Pick<ButtonProps, "size" | "isOutlined">>`
  ${({ theme, size, isOutlined }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    padding: ${size === "normal" ? "12px 40px" : "9px 30px"};
    border-radius: 5px;
    border: none;
    font-family: ${theme.font.family};
    font-size: 1.5rem;
    letter-spacing: 0.5px;
    cursor: pointer;
    transition: all 0.3s ease;

    ${isOutlined && buttonModifiers.isOutlined(theme)}

    &:hover {
      background-color: ${theme.colors.secondary};
      transform: translateY(-3px);
    }
  `}
`;
