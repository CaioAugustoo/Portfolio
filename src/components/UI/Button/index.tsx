import * as S from "./styles";

import { ButtonProps } from "types/types";

const Button = ({ fullWidth = false, children }: ButtonProps) => {
  return <S.Button fullWidth={fullWidth}>{children}</S.Button>;
};

export default Button;
