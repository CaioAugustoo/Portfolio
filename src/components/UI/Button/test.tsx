import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/tests/helpers";

import Button from ".";

describe("<Button />", () => {
  it("should render with children", () => {
    const { container } = renderWithTheme(<Button>Button</Button>);

    expect(screen.getByText(/Button/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render with some styles", () => {
    renderWithTheme(<Button>Button</Button>);

    expect(screen.getByRole("button", { name: /Button/i })).toHaveStyle({
      "background-color": "#4C30F5",
      padding: "12px 40px",
      "font-size": "1.5rem",
    });
  });
});
