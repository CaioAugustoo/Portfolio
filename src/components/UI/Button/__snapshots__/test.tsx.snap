// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render with children 1`] = `
.c0 {
  background-color: #4C30F5;
  color: #ffffff;
  padding: 12px 40px;
  border-radius: 5px;
  border: none;
  font-family: Poppins,sans-serif;
  font-size: 1.5rem;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  cursor: pointer;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c0:hover {
  background-color: #7962ff;
  -webkit-transform: translateY(-3px);
  -ms-transform: translateY(-3px);
  transform: translateY(-3px);
}

<button
  class="c0"
>
  Button
</button>
`;
